VmAgent.Core.UnitTets/SystemOperationsTests.cs

```diff
@@ -0,0 +1,293 @@
+                    {
+                        toReturn.Add(subDirectoryFileInfo);
+                        toReturn.Add(directoryFileInfo);
+                    }
+                    return toReturn;
+                });
+
+            _mockFileSystemOperations.Setup(x => x.GetFiles(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _directoryPath)), It.IsAny<bool>()))
+                .Returns<DirectoryInfo, bool>(
+                (dirInfo, recursive) =>
+                {
+                    List<FileInfo> toReturn = new List<FileInfo>();
+                    toReturn.Add(directoryFileInfo);
+                    if (recursive)
+                    {
+                        toReturn.Add(subDirectoryFileInfo);
+                    }
+                    return toReturn;
+                });
+
+            _mockFileSystemOperations.Setup(x => x.GetFiles(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _subdirectoryPath)), It.IsAny<bool>()))
+               .Returns<DirectoryInfo, bool>(
+                (dirInfo, recursive) =>
+                {
+                    List<FileInfo> toReturn = new List<FileInfo>();
+                    toReturn.Add(subDirectoryFileInfo);
+                    return toReturn;
+                });
+
+            _mockFileSystemOperations.Setup(x => x.GetDirectories(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _root)), It.IsAny<bool>()))
+               .Returns<DirectoryInfo, bool>(
+               (dirInfo, recursive) =>
+               {
+                   List<DirectoryInfo> toReturn = new List<DirectoryInfo>();
+                   toReturn.Add(_directoryInfo);
+                   if (recursive)
+                   {
+                       toReturn.Add(_subDirectoryInfo);
+                   }
+                   return toReturn;
+               });
+
+            _mockFileSystemOperations.Setup(x => x.GetDirectories(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _directoryPath)), It.IsAny<bool>()))
+                .Returns<DirectoryInfo, bool>(
+                (dirInfo, recursive) =>
+                {
+                    List<DirectoryInfo> toReturn = new List<DirectoryInfo>();
+                    toReturn.Add(_subDirectoryInfo);
+                    return toReturn;
+                });
+
+            _mockFileSystemOperations.Setup(x => x.GetDirectories(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _subdirectoryPath)), It.IsAny<bool>()))
+               .Returns<DirectoryInfo, bool>(
+                (dirInfo, recursive) =>
+                {
+                    List<DirectoryInfo> toReturn = new List<DirectoryInfo>();
+                    return toReturn;
+                });
+
+            _mockFileSystemOperations.Setup(x => x.GetFileSystemInfos(It.Is<DirectoryInfo>((info) => IsFileSystemInfoEqual(info, _root)), It.IsAny<bool>()))
+               .Returns<DirectoryInfo, bool>(
+               (dirInfo, recursive) =>
+               {
```

Comment:
This is very long, consider breaking it up into method so 1) its easier to follow and 2) anyone diving into this in the future can navigate quicker